### PR TITLE
feat: use `logkitty` to format android logs

### LINKED ISSUE #125: RFC: use logkitty for `log-android` command
## Describe the Feature

This is a proposal to use 3rd party tool for `log-android` for prettier messages and more flexible interface: https://github.com/zamotany/logkitty

logkitty

logkitty currently supports only Android (iOS in the future if requested) with better syntax for common tasks like filtering by tags (`logkitty tag myTag otherTag`, `logkitty tag myTag otherTag -W`), filtering by application (`logkitty app com.expample.myApp`, `logkitty app com.expample.myApp -E`) and custom patters (`logkitty custom *:S ReactNativeJS:D`).

Let me know what do you think. Any feedback appreciated.

## Possible Implementations

Spawn `logkitty` process when running `react-native log-android`.

### PR DESCRIPTION
**Pull request by Esemesek**
Summary:
---------

Use `logkitty` to format android logs.

Fixes #125 

cc @zamotany

### CODE REVIEW

**Comment by zamotany** [comment]
`ansi-fragments@0.2.1` released along side `logkitty@0.3.2`, you are good to go

File: packages/cli/src/commands/logAndroid/logAndroid.js [hunk 1]
```diff
@@ -5,26 +5,31 @@
...
 
-  logger.info(`Starting the logger (${adbPath} ${adbArgs.join(' ')})...`);
+  const emitter = logkitty({
+    platform: 'android',
+    minPriority: Priority.VERBOSE,
+    filter: makeTagsFilter('ReactNative', 'ReactNativeJS'),
+  });
 
-  const log = spawnSync(adbPath, adbArgs, {stdio: 'inherit'});
+  emitter.on('entry', (entry: Entry) => {
```
Comment (thymikee) [inline packages/cli/src/commands/logAndroid/logAndroid.js]: Entry type is not imported but I guess there's no Flow types for this lib

Comment (thymikee) [reply packages/cli/src/commands/logAndroid/logAndroid.js]: also does `entry` event cover all levels of logs?

Comment (Esemesek) [reply packages/cli/src/commands/logAndroid/logAndroid.js]: As far as I know, `entry covers it all, except `error`. There are no Flow types for this lib, but for some reason, `Flow` is assuming `any` on undefined types. I'll change it.

Comment (zamotany) [reply packages/cli/src/commands/logAndroid/logAndroid.js]: https://github.com/zamotany/logkitty/blob/master/src/types.ts#L1-L7

File: packages/cli/src/commands/logAndroid/logAndroid.js [hunk 2]
```diff
@@ -5,26 +5,31 @@
...
-  const adbPath = process.env.ANDROID_HOME
-    ? `${process.env.ANDROID_HOME}/platform-tools/adb`
-    : 'adb';
-
-  const adbArgs = ['logcat', '*:S', 'ReactNative:V', 'ReactNativeJS:V'];
+  logger.info('Starting logkitty');
 
-  logger.info(`Starting the logger (${adbPath} ${adbArgs.join(' ')})...`);
+  const emitter = logkitty({
+    platform: 'android',
```
Comment (grabbou) [inline packages/cli/src/commands/logAndroid/logAndroid.js]: Interesting. That means we could possibly support iOS in the future too!

**Comment by grabbou** [comment]
BTW @Esemesek, mind adding a screenshot of a new output to the PR description so that we have it for future reference?

**Pull request merged**